### DISCUSSION TITLE
Shared Entities project for Client and Server not possible?

### DISCUSSION THREAD

**Discussion by sascha-yy** [Azure/azure-mobile-apps#698] [Q&A] [answered]
Is there any way to create a shared entities project, which can be used by client and server?

Currently, the entities need inheritance either from `EntityTableData` or `DatasyncClientData`.

The major difference is the `Version`-property, as it has a completely different type:
- on server its `byte[]` 
- on client its `string`

There is no "IEntityBase"-interface that can be used for both: client and server.
Also, on the server side it has to be the implementation of `EntityTableData` (not an interface), since it is validated by this type in the controllers, if i am not mistaken.

Is there any solution for this that i have overlooked?

**Comment by adrianhall** [comment, marked as answer]
No, there isn't.  There are good reasons why not (the entities on the server require attributes that are not available on the client to do their work).
